plugins/modules/ibm_resource_instance.py

```diff
@@ -1,115 +1,187 @@
 
-# (C) Copyright IBM Corp. 2022.
+# (C) Copyright IBM Corp. 2023.
+#
 # GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
 
-# pylint: disable=missing-function-docstring,too-many-branches
-
-
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
 
 DOCUMENTATION = r'''
 ---
 module: ibm_resource_instance
-short_description: Manage ibm_resource_instance resources.
-author: Kavya Handadi (@kavya498)
-version_added: "0.0.1-beta0"
+short_description: Manage C(resource_instances) for Resource Controller.
+author:
+  - Kavya Handadi (@kavya498)
+  - Umar Ali (@umarali-nagoor)
+version_added: "1.0.0"
 description:
-    - This module creates, updates, or deletes a ibm_resource_instance.
-    - By default the module will look for an existing ibm_resource_instance.
+  - This module creates, updates, or deletes a C(resource_instance) resource for Resource Controller.
 requirements:
-    - "ResourceControllerV2"
+  - "ResourceControllerV2"
 options:
-    resource_group:
-        description:
-            - The ID of the resource group.
-        type: str
-    plan:
-        description:
-            - The unique ID of the plan associated with the offering. This value is provided by and stored in the global catalog.
-        type: str
-    allow_cleanup:
-        description:
-            - A boolean that dictates if the resource instance should be deleted (cleaned up) during the processing of a region instance delete call.
-        type: bool
-    name:
-        description:
-            - The name of the instance. Must be 180 characters or less and cannot include any special characters other than `(space) - . _ :`.
-        type: str
-    parameters:
-        description:
-            - Configuration options represented as key-value pairs that are passed through to the target resource brokers.
-        type: dict
-    location:
-        description:
-            - The deployment location where the instance should be hosted.
-        type: str
-    tags:
-        description:
-            - Tags that are attached to the instance after provisioning. These tags can be searched and managed through the Tagging API in IBM Cloud.
-        type: list
-        elements: str
-    entity_lock:
-        description: |
-            Indicates if the resource instance is locked for further update or delete operations.
-            It does not affect actions performed on child resources like aliases, bindings or keys. False by default.
-        type: bool
-    id:
-        description:
-            - The ID of the instance.
-        type: str
-    recursive:
-        description:
-            - Will delete resource bindings, keys and aliases associated with the instance.
-        type: bool
-    state:
-        description:
-            - Should the resource be present or absent.
-        type: str
-        default: present
-        choices: [present, absent]
+  resource_group:
```

Comment:
In the first glance, resource_instance module documentation section doesn’t look good to me..
It has to match with the args that is defined in run_module function.
Can you please make sure it is maintained in all other modules also?